### PR TITLE
e2e: Increases the memory limit for downward API tests

### LINKED ISSUE #95226: [Flaky Test] [sig-storage] Downward API volume should provide container's memory limit [NodeConformance] [Conformance]
**Which jobs are flaking**:
node-kubelet-master
**Which test(s) are flaking**:
[sig-storage] Downward API volume should provide container's memory limit [NodeConformance] [Conformance] [cos-stable1]
[sig-storage] Downward API volume should provide container's cpu limit [NodeConformance] [Conformance] [ubuntu]

**Testgrid link**:
https://k8s-testgrid.appspot.com/sig-release-master-blocking#node-kubelet-master&width=20
**Reason for failure**:

Both of these tests are failing with same reason.

```
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:629
Unexpected error:
    <*errors.errorString | 0xc001284970>: {
        s: "expected pod \"downwardapi-volume-7241f623-b626-474e-a516-63a30ffccf34\" success: pod \"downwardapi-volume-7241f623-b626-474e-a516-63a30ffccf34\" failed with status: {Phase:Failed Conditions:[] Message:Pod Node didn't have enough resource: cpu, requested: 250, used: 1000, capacity: 1000 Reason:OutOfcpu NominatedNodeName: HostIP: PodIP: PodIPs:[] StartTime:2020-09-30 23:27:58 +0000 UTC InitContainerStatuses:[] ContainerStatuses:[] QOSClass: EphemeralContainerStatuses:[]}",
    }
    expected pod "downwardapi-volume-7241f623-b626-474e-a516-63a30ffccf34" success: pod "downwardapi-volume-7241f623-b626-474e-a516-63a30ffccf34" failed with status: {Phase:Failed Conditions:[] Message:Pod Node didn't have enough resource: cpu, requested: 250, used: 1000, capacity: 1000 Reason:OutOfcpu NominatedNodeName: HostIP: PodIP: PodIPs:[] StartTime:2020-09-30 23:27:58 +0000 UTC InitContainerStatuses:[] ContainerStatuses:[] QOSClass: EphemeralContainerStatuses:[]}
occurred
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:732
```

**Anything else we need to know**:
- links to go.k8s.io/triage appreciated
https://storage.googleapis.com/k8s-gubernator/triage/index.html?test=Downward%20API%20volume%20should%20provide%20container%27s%20memory%20limit

```

Time | Job | Test
-- | -- | --
01/10/2020, 06:04:01 | ci-kubernetes-e2e-windows-containerd-gce-1-18 | [sig-storage] Downward API volume should provide container's memory limit [NodeConformance] [Conformance]
01/10/2020, 01:48:10 | ci-kubernetes-e2e-windows-containerd-gce-1-18 | [sig-storage] Downward API volume should provide container's memory limit [NodeConformance] [Conformance]
30/09/2020, 17:46:43 | ci-kubernetes-e2e-windows-containerd-gce-1-18 | [sig-storage] Downward API volume should provide container's memory limit [NodeConformance] [Conformance]
30/09/2020, 05:40:57 | ci-kubernetes-e2e-windows-containerd-gce-1-18 | [sig-storage] Downward API volume should provide container's memory limit [NodeConformance] [Conformance]
30/09/2020, 01:38:53 | ci-kubernetes-e2e-windows-containerd-gce-1-18 | [sig-storage] Downward API volume should provide container's memory limit [NodeConformance] [Conformance]


```
- links to specific failures in spyglass appreciated
https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-kubernetes-node-kubelet/1311445687300263936

/sig storage

### PR DESCRIPTION
**Pull request by claudiubelu**
#### What type of PR is this?

/kind flake
/sig windows
/sig testing

#### What this PR does / why we need it:

Currently, the downward API tests flake on Windows with a failure to allocate memory when starting the ``agnhost`` binary used in these tests. The tests are spawning pods with a memory limit of 64MB, which is a bit on the low side for a Windows Pod, even if it's a nanoserver-based image.

Increases the memory limit to 128MB, the primary goal of the tests is not to enforce and test the limits, but to check if these details are projected into the Pod.

#### Which issue(s) this PR fixes:

Partially-Fixes: #95226

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by claudiubelu** [comment]
A bit more context for the issue we're trying to solve: https://github.com/kubernetes/kubernetes/issues/95226#issuecomment-1766852887

File: test/e2e/common/storage/downwardapi_volume.go [hunk 1]
```diff
@@ -339,7 +339,7 @@ func downwardAPIVolumeBaseContainers(name, filePath string) []v1.Container {
 				},
 				Limits: v1.ResourceList{
 					v1.ResourceCPU:    resource.MustParse("1250m"),
-					v1.ResourceMemory: resource.MustParse("64Mi"),
+					v1.ResourceMemory: resource.MustParse("128Mi"),
```
Comment (jsturtevant) [inline test/e2e/common/storage/downwardapi_volume.go:342]: is this value checked later in the tests?

Comment (marosset) [reply test/e2e/common/storage/downwardapi_volume.go:342]: Yes

```log
   <*errors.errorString | 0xc000f16a30>: 
      expected "67108864\n" in container output: Expected
          <string>: content of file "/etc/podinfo/memory_limit": 134217728
          
      to contain substring
          <string>: 67108864
```

**Comment by jsturtevant** [comment]
looks like a few lints need to be fixed but otherwise looks good

**Review by marosset — APPROVED** [review]
/lgtm

**Comment by hailkomputer** [comment]
Hello, bug triage team lead here. I want to check the status and remind that the the code freeze is starting [01:00 UTC Wednesday 1st November 2023 / 18:00 PDT Tuesday 31st October 2023](https://everytimezone.com/s/24b27b03) (next week), as we want to ensure that each PR has a chance to be merged.
Are we still targeting the 1.29 release?

**Pull request merged**